### PR TITLE
Properly reap zombie processes

### PR DESCRIPTION
**Pull request by chriskuehl**
~~~Still TODO: tests~~~

### CODE REVIEW

File: dumb-init.c [hunk 1]
```diff
@@ -23,21 +24,46 @@
...
     }
 }
 
+void reap_zombies(int signum) {
+    /*
+     * As PID 1, dumb-init is expected to handle reaping of zombie processes.
+     *
+     * If a process's parent exits, the child is orphaned and its new parent is
+     * PID 1. If that child later exits, it becomes a zombie process until its
+     * parent (now dumb-init) calls wait() on it.
```
Comment (kentwills) [inline dumb-init.c]: note: we don't simply use SIGDFL because we want the exit code of our direct subprocess.

**Comment by chriskuehl** [comment]
This is blocked by #13 since we can't test it without Docker

**Comment by kentwills** [comment]
unblocked now.

**Comment by chriskuehl** [comment]
@bukzor @kentwills This should be ready to go. All the tests passed on Travis except the flaky one which I'll work on tomorrow (only one itest target failed). I'm re-running it.

**Comment by chriskuehl** [comment]
succeeded on re-run

**Pull request merged**